src/tallies/filter_sptl_legendre.cpp

```diff
@@ -0,0 +1,209 @@
+#include "openmc/tallies/filter_sptl_legendre.h"
+
+#include <utility>  // For pair
+
+#include "openmc/capi.h"
+#include "openmc/error.h"
+#include "openmc/math_functions.h"
+#include "openmc/xml_interface.h"
+
+namespace openmc {
+
+void
+SpatialLegendreFilter::from_xml(pugi::xml_node node)
+{
+  this->set_order(std::stoi(get_node_value(node, "order")));
+
+  auto axis = get_node_value(node, "axis");
+  switch (axis[0]) {
+  case 'x':
+    this->set_axis(LegendreAxis::x);
+    break;
+  case 'y':
+    this->set_axis(LegendreAxis::y);
+    break;
+  case 'z':
+    this->set_axis(LegendreAxis::z);
+    break;
+  default:
+    throw std::runtime_error{"Axis for SpatialLegendreFilter must be 'x', 'y', or 'z'"};
+  }
+
+  double min = std::stod(get_node_value(node, "min"));
+  double max = std::stod(get_node_value(node, "max"));
+  this->set_minmax(min, max);
+}
+
+void
+SpatialLegendreFilter::set_order(int order)
+{
+  if (order < 0) {
+    throw std::invalid_argument{"Legendre order must be non-negative."};
+  }
+  order_ = order;
+  n_bins_ = order_ + 1;
+}
+
+void
+SpatialLegendreFilter::set_axis(LegendreAxis axis)
+{
+  axis_ = axis;
+}
+
+void
+SpatialLegendreFilter::set_minmax(double min, double max)
+{
+  if (max < min) {
```

Comment:
This should probably be `max <= min`, because we divide by `max-min`, so if `max==min` we will have an error in our calculations.